cmd/prompt.go

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	prompt "github.com/c-bata/go-prompt"
+	"github.com/kris-nova/kubicorn/cutil/logger"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var commandFlagSuggestions = make(map[string][]prompt.Suggest)
+var commandMap = make(map[string][]prompt.Suggest)
+var promptSuggestions = []prompt.Suggest{}
+var promptCmd = &cobra.Command{
+	Use:   "prompt",
+	Short: "Open a prompt with auto-completion (non-Windows)",
+	Long: `Use this command to use the Kubicron API via a shell prompt.
+
+This command will open a prompt using go-prompt (with auto-completion) to
+allow you to run commands interactively from the shell.
+Currently this doesn't work on Windows systems`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) > 0 {
+			logger.Critical("Too many arguments.")
+			os.Exit(1)
+		}
+		err := RunPrompt()
+		if err != nil {
+			logger.Critical(err.Error())
+			os.Exit(1)
+		}
+	},
+}
+
+func RunPrompt() error {
+	fmt.Println("Please input your kubicorn commands.")
+	defer fmt.Println("Bye!")
+
+	p := prompt.New(
+		executor,
+		completer,
+		prompt.OptionTitle("kubicorn: interactive kubicorn client"),
+		prompt.OptionPrefix(">> "),
+		prompt.OptionInputTextColor(prompt.Yellow))
```

Comment:
Let's get get rid of setting input text color. This is not user-friendly for users using bright colors for terminal (e.g. users with white terminal will hardly see anything).

Without this, terminal will automatically handle the color.
I'm also open to another solution if you have.

Also, I prefer putting parenthesis in the new line, to improve readability, e.g.:
```go
p := prompt.New(
 		executor,
 		completer,
 		prompt.OptionTitle("kubicorn: interactive kubicorn client"),
 		prompt.OptionPrefix(">> "),
)
```